### ISSUE TITLE
Backpropagation pipeline

### ISSUE THREAD

**Issue by JacobGlennAyers** [JacobGlennAyers/CNN#4]
- I am thinking that the loss function should automatically be weighted to be as general as possible with minimum labor.
